py_zipkin/storage.py

```diff
@@ -70,5 +78,42 @@ def set_transport_configured(self, configured):
         self._is_transport_configured = configured
 
 
-def default_span_storage():
-    return thread_local.get_thread_local_span_storage()
+class ThreadLocalSpanStorage(object):
```

Comment:
hmm. I wish this was of type `SpanStorage`. What if this subclasses `SpanStorage`, and `SpanStorage` can be a base class that looks kind of like a deque, but with only the methods/properties that we need. That way if people want to create their own subtypes, what they need to implement is clear.

I think this would also allow you to move the implementations to the base class and then the subclass would only really need to implement the `storage` property.

Reply:
At first I made this a subclass of `SpanStorage`. The issue then was that it was hard to tell if I had forgotten to override some method. Being a subclass of `object` means that if I forget a method it'll just throw an error.

Not a big deal though now that it's implemented though...

Reply:
I think @bplotnick raised a very good point about the common type between the implementations. Have we considered to create an abstract class for the span storage? In that case it will force to implement the required methods to make it functional.

Reply:
@signalpillar Can you share an example of how you implemented it? I've no idea how people would implement their own SpanStorage object and what requirements they have, so I'm unsure on what's the best API.

My original idea was that it should be enough instantiate a normal `SpanStorage`, pass it around between functions or store it in your won temporary storage and then just pass it to every `with zipkin_span`. But maybe that's not how you actually use it / there's a better way.

Reply:
I am sorry @drolando, I missed the point that stack and storage are completely different things. We use them differently and they are not related. I needed to implement different stack. I believe we didn't have storage back then. It is quite new thing.

I still believe that it is better to have storage base class as for the different runtime implementation like in the case with eventlet we will need a different implementation.

Is it correct that our current abstract class should look like the following?

```python
class AbstractSpanStorage(abc.ABC):

    @abc.abstractmethod
    def is_transport_configured(self):
        ...

    @abc.abstractmethod
    def set_transport_configured(self, configured):
        ...

    @abc.abstractmethod
    def __iter__(self):
        ...

    @abc.abstractmethod
    def __len__(self):
        ...

    @abc.abstractmethod
    def append(self, el):
        ...

    @abc.abstractmethod
    def clear(self):
        ...
```